src/protocol/prss.rs

```diff
@@ -161,6 +165,7 @@ impl IndexedSharedRandomness {
 pub struct SequentialSharedRandomness {
     generator: Generator,
     counter: u128,
+    _span: EnteredSpan,
```

Comment:
`SequentialSharedRandomness` API supports spans out of the box, no need to change context API for it. It does not have to be `Send` (it is not within an `Arc`), so it can just carry the span (with step/role fields attached)